src/EnergyPlus/ScheduleManager.cc

```diff
@@ -2245,47 +2223,23 @@ namespace ScheduleManager {
         // PURPOSE OF THIS SUBROUTINE:
         // This subroutine puts the details of the Schedules on the .eio file (Inits file).
 
-        // METHODOLOGY EMPLOYED:
-        // na
-
-        // REFERENCES:
-        // na
-
-        // Using/Aliasing
-
-        // Locals
-        // SUBROUTINE ARGUMENT DEFINITIONS:
-
         // SUBROUTINE PARAMETER DEFINITIONS:
-        Array1D_string const Months(12, {"Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"});
-        Array1D_string const HrField({0, 24}, {"00", "01", "02", "03", "04", "05", "06", "07", "08", "09", "10", "11", "12",
-                                               "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24"});
-
-        // INTERFACE BLOCK SPECIFICATIONS
-        // na
-
-        // DERIVED TYPE DEFINITIONS
-        // na
+        constexpr std::array<std::string_view, 12> Months = {"Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"};
+        constexpr std::array<std::string_view, 25> HrField = {"00", "01", "02", "03", "04", "05", "06", "07", "08", "09", "10", "11", "12",
+                                                              "13", "14", "15", "16", "17", "18", "19", "20", "21", "22", "23", "24"};
 
         // SUBROUTINE LOCAL VARIABLE DECLARATIONS:
-        int Count;
-        int Hr;
-        int TS;
         int NumF;
         int PMon;
         int PDay;
-        int iWeek;
-        int iDay;
-        int DT;
-        int iDayP;
         Array1D_string ShowMinute;
-        int CurMinute;
         Array1D_string TimeHHMM;
         std::string NoAverageLinear;
         std::string YesNo2;
         std::string Num1;
         std::string Num2;
         Array2D_string RoundTSValue;
+        auto constexpr SchDFmtdata{",{}"};
```

Comment:
prefer `string` or `string_view` here, whichever you need. Not sure what `auto` does.

Reply:
String view it is!